openmc/tallies.py

```diff
@@ -1468,32 +1528,20 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Query the mean and std dev so the tally data is read in from file
+        # if it has not already been read in.
+        self.mean, self.std_dev, other.mean, other.std_dev
```

Comment:
I am in favor of adding a `load_tally_data(...)` and/or `clear_tally_data(...)` method and having this called in the `statepoint.get_tally(...)` method. I will also note that line 339 of `tallies.py` performs a force read of `sum` data, so adding in the methods would benefit other code blocks as well. My opinion is that we keep this as is in this PR and create an issue for this to be handled in a future PR. Is this agreeable?